exonum-light-client/src/main/java/com/exonum/client/ExonumClient.java

```diff
@@ -17,9 +17,90 @@
 
 package com.exonum.client;
 
+import com.exonum.binding.common.hash.HashCode;
+import com.exonum.binding.common.message.TransactionMessage;
+import java.net.MalformedURLException;
+import java.net.URL;
+import okhttp3.OkHttpClient;
+
 /**
- * Exonum client main class to start with.
- */
+ * Main class to start using Exonum Light client.
+ * Provides a convenient way for interaction with Exonum framework APIs.
+ * It uses {@linkplain OkHttpClient} internally for REST API communication with Exonum node.
+ **/
 public interface ExonumClient {
-  //TODO: implement
+
+  /**
+   * Submits the transaction message to an Exonum node.
+   * @return transaction message hash
+   * @throws RuntimeException if network request fails
+   */
+  HashCode submitTransaction(TransactionMessage tx);
+
+  /**
+   * Returns Exonum client builder.
+   */
+  static Builder newBuilder() {
+    return new Builder();
+  }
+
+  /**
+   * Builder class for the Exonum client.
+   */
+  class Builder {
+    private static final OkHttpClient DEFAULT_CLIENT = new OkHttpClient();
+
+    private URL exonumHost;
+    private OkHttpClient httpClient = DEFAULT_CLIENT;
+
+    /**
+     * Sets Exonum host url.
+     */
+    public Builder setExonumHost(URL exonumHost) {
+      this.exonumHost = exonumHost;
```

Comment:
Shall we do anything to validate it? Not null? What will happen in case of more subtle errors (e.g., wrong port, wrong protocol)? An exception in `submit…`?

Reply:
- there is the format validation inside the `URL` (that's because I use it)
- we validate for `null` at `build()`. But anyway it's ok to have `chechNotNull` as well